src/core/execute.c

```diff
@@ -453,6 +453,12 @@ int exec_spawn(
          * handoff timestamp. */
         dual_timestamp_now(&start_timestamp);
 
+        r = 0;
+        if (unit->manager->starting_ambient_set != 0 && MANAGER_IS_USER(unit->manager))
+                r = capability_ambient_set_apply(unit->manager->starting_ambient_set, /* also_inherit= */ false);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Failed to apply the starting ambient set: %m");
```

Comment:
just include these two lines in the earlier if condition? that way you don't have to do the `r = 0` thing either

Reply:
there should be a comment here explaining the situation. this is not obvious at all otherwise

Reply:
I don't like this approach at all, it's too complex for a just-in-case. As already mentioned, we should just add a drop caps flag in safe_fork, and use it for generators. Simpler, safer, fine to merge after rc2.

Reply:
see my comment below: i think it's safer to run without ambient caps and re-acquire them where necessary (which is an absolutely exceptional case btw), rather than trying the opposite.